quantum/audio/audio_avr_pwm.c

```diff
@@ -0,0 +1,307 @@
+/* Copyright 2016 Jack Humbert
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <string.h>
+//#include <math.h>
+#if defined(__AVR__)
+  #include <avr/pgmspace.h>
+  #include <avr/interrupt.h>
+  #include <avr/io.h>
+#endif
+#include "print.h"
+#include "audio.h"
+
+#include "wait.h"
+
+//TODO: move into audio-common state
+extern bool playing_notes;
+extern float note_timbre;
+extern uint8_t voices;
+
+#define CPU_PRESCALER 8
+
+// -----------------------------------------------------------------------------
+// Timer Abstractions
+// -----------------------------------------------------------------------------
+
+//Currently we support timers 1 and 3 used at the same time, channels A-C,
+//pins PB5, PB6, PB7, PC4, PC5, and PC6
+
+#if defined(B5_AUDIO) || defined(B6_AUDIO) || defined(B7_AUDIO)
+    #define AUDIO1_PIN_SET
+    #define AUDIO1_TIMSKx TIMSK1
+    #define AUDIO1_DDRx DDRB
+    #define AUDIO1_TCCRxA TCCR1A
+    #define AUDIO1_TCCRxB TCCR1B
+    #define AUDIO1_ICRx ICR1
+    #define AUDIO1_WGMx0 WGM10
```

Comment:
Technically, we can just use the Timer 1 defines for `WGMxx` and `CSxx`, since the `TCCR` registers are identical. `WGM10` and `WGM30` will evaluate to the same bit position.

Reply:
err, sorry - no. that would be too misleading :-(
(atmel probably hat a good reason for differentiating the those bit-offsets)

Reply:
I'll grant you that, for example, the `WGM20` bit of `TCCR2` on the ATmega32A is bit 6 compared to `WGM10` on the ATmega32U4 being bit 0, however since this code currently only supports Timers 1 and 3 (of which only the former is present & matches on the 32A), I don't think it really matters all that much.

Reply:
ok - leaving the defines as they are :-)